### DISCUSSION TITLE
[Errno 95] Operation not supported

### DISCUSSION THREAD

**Discussion by madiltalay** [open-mmlab/mmdetection#8729] [General] [answered]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
While running `train.py`, the epoch runs, but when the time to save checkpoint is reached, I get:

>OSError                                   Traceback (most recent call last)
<ipython-input-24-9392f7469087> in <module>()
----> 1 train_detector(model, datasets, cfg, distributed=distributed, validate=validate, timestamp=timestamp, meta=meta)
7 frames
/content/packages/mmcv/utils/path.py in symlink(src, dst, overwrite, **kwargs)
     36     if os.path.lexists(dst) and overwrite:
     37         os.remove(dst)
---> 38     os.symlink(src, dst, **kwargs)
     39 
     40 
OSError: [Errno 95] Operation not supported: 'epoch_1.pth' -> '/content/gdrive/My Drive/Developer/Project_1/checkpoints/resnet_retina/epoch_1.pth'

Checking the issues, I find that I have to set `create_symlink=False` in runner.py in mmcv.runner.
I set `create_symlink=False` and rerun the code. Then the epoch does not run and I get the error:

>RuntimeError: cuda runtime error (209) : no kernel image is available for execution on the device at mmdet/ops/sigmoid_focal_loss/src/cuda/sigmoid_focal_loss_cuda.cu:128

I don't understand what am I doing wrong.
**Reproduction**

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use?

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment infomation and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.
```
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by madiltalay** [comment]
Well it seems that I cannot use symlink twice, and as I was using it earlier in the script, the second time usage gave an error. I found a workaround to avoid the usage in the first instance which led to error-free running. 
However, it is still strange that why does setting `create_symlink = False` leads to cuda error.
If that's a bug, it might be fixed, else the issue maybe closed.
Thanks.

**Comment by hellock** [comment]
Thanks for reporting and we will check it.

**Comment by ProBuro** [comment]
how did you solve the problem? I have the same in colab

**Comment by ProBuro** [comment, marked as answer]
I found a solution to the problem. there it says: replace with create_symlink = False. strange that no one has yet answered this question. / usr / ... /mmcv/runner/epoch_based_runner.py
works in colab
